Sources/AWSLambdaRuntime/Lambda+Codable.swift

```diff
@@ -78,6 +78,70 @@ internal struct CodableVoidClosureWrapper<In: Decodable>: LambdaHandler {
     }
 }
 
+// MARK: - Async
+
+#if compiler(>=5.5) && $AsyncAwait
+extension Lambda {
+    /// An async Lambda Closure that takes a `In: Decodable` and returns an `Out: Encodable`
+    public typealias CodableAsyncClosure<In: Decodable, Out: Encodable> = (Lambda.Context, In) async throws -> Out
+
+    /// Run a Lambda defined by implementing the `CodableAsyncClosure` function.
+    ///
+    /// - parameters:
+    ///     - closure: `CodableAsyncClosure` based Lambda.
+    ///
+    /// - note: This is a blocking operation that will run forever, as its lifecycle is managed by the AWS Lambda Runtime Engine.
+    public static func run<In: Decodable, Out: Encodable>(_ closure: @escaping CodableAsyncClosure<In, Out>) {
+        self.run(CodableAsyncWrapper(closure))
+    }
+
+    /// An asynchronous Lambda Closure that takes a `In: Decodable` and returns nothing.
+    public typealias CodableVoidAsyncClosure<In: Decodable> = (Lambda.Context, In) async throws -> Void
```

Comment:
I would advise against typealiases for closures, they give you really bad completions and it's unclear to the user what they're supposed to put there (because they'll just see `CodableVoidAsyncClosure<In>`).

Reply:
I support your argument. 👍 
However if we don't use a `typealias` here, we should later adjust all other the callback closures as well. @tomerd wdyt?